potassium/potassium.py

```diff
@@ -1,70 +1,70 @@
 import time
+import os
 from types import GeneratorType
-from typing import Generator, Optional, Union
 from flask import Flask, request, make_response, abort, Response as FlaskResponse
+from huggingface_hub.file_download import uuid
 from werkzeug.serving import make_server
-from werkzeug.datastructures.headers import EnvironHeaders
-from threading import Thread, Lock, Condition
+from threading import Thread, Lock
+from queue import Queue as ThreadQueue
 import functools
-import traceback
-import json as jsonlib
 from termcolor import colored
-
-
-class Endpoint():
-    def __init__(self, type, func):
-        self.type = type
-        self.func = func
-
-class Request():
-    def __init__(self, id: str, headers: EnvironHeaders, json: dict):
-        self.id = id
-        self.headers = headers
-        self.json = json
-
-ResponseBody = Union[bytes, Generator[bytes, None, None]]
-
-class Response():
-    def __init__(self, status: int = 200, json: Optional[dict] = None, headers: Optional[dict] = None, body: Optional[ResponseBody] = None):
-        assert json == None or body == None, "Potassium Response object cannot have both json and body set"
-
-
-        self.headers = headers if headers != None else {}
-
-        # convert json to body if not None
-        if json != None:
-            self.body = jsonlib.dumps(json).encode("utf-8")
-            self.headers["Content-Type"] = "application/json"
-        else:
-            self.body = body
-
-        self.status = status
-
-    @property
-    def json(self):
-        if self.body == None:
-            return None
-        if type(self.body) == bytes:
-            try:
-                return jsonlib.loads(self.body.decode("utf-8"))
-            except:
-                return None
-        return None
-            
-    @json.setter
-    def json(self, json):
-        self.body = jsonlib.dumps(json).encode("utf-8")
-        self.headers["Content-Type"] = "application/json"
-
-
-class InvalidEndpointTypeException(Exception):
-    def __init__(self):
-        super().__init__("Invalid endpoint type. Must be 'handler' or 'background'")
-
-
-class RouteAlreadyInUseException(Exception):
-    def __init__(self):
-        super().__init__("Route already in use")
+from multiprocessing import Pool as ProcessPool, Queue as ProcessQueue
+from multiprocessing.pool import ThreadPool
+from .status import PotassiumStatus, StatusEvent
+from .worker import run_worker, init_worker
+from .exceptions import RouteAlreadyInUseException, InvalidEndpointTypeException
+from .types import Request, Endpoint, RequestHeaders, Response
+
+class ResponseMailbox():
+    def __init__(self, response_queue):
+        self._response_queue = response_queue
+        self._mailbox = {}
+        self._lock = Lock()
+
+        t = Thread(target=self._response_handler, daemon=True)
+        t.start()
```

Comment:
if using a variable with a name like `t` why not just do Thread(...).start() ? I don't see other usage.